### PR TITLE
17-11 Security Update

### PR DESCRIPTION
**Pull request by leirocks**
17-11 Security Update that addresses the following issues in ChakraCore

CVE-2017-11791
CVE-2017-11836
CVE-2017-11837
CVE-2017-11838
CVE-2017-11840
CVE-2017-11841
CVE-2017-11843
CVE-2017-11846
CVE-2017-11858
CVE-2017-11861
CVE-2017-11862
CVE-2017-11870
CVE-2017-11871
CVE-2017-11873
CVE-2017-11874
CVE-2017-11866
CVE-2017-11859

### CODE REVIEW

**Comment by leirocks** [comment]
fixed "BackEnd.h" with "Backend.h"

File: lib/Runtime/Language/JavascriptOperators.cpp [hunk 1]
```diff
@@ -4772,64 +4772,45 @@ namespace Js
         return returnValue;
     }
 
+    template<typename T, typename T(*func)(Var, ScriptContext*)> bool MemsetConversion(Var value, ScriptContext* scriptContext, T* result)
```
Comment (MSLaguana) [inline lib/Runtime/Language/JavascriptOperators.cpp]: clang is complaining about this line, it doesn't seem to like the `typename T(*func)(Var, ScriptContext*)`

```
[1m/mnt/j/w/Microsoft_ChakraCore/release_1.7/shared_ubuntu_linux_debug_prtest/lib/Runtime/Language/JavascriptOperators.cpp:4775:35: [0m[0;1;31merror: [0m[1mexpected a qualified name after 'typename' [-Werror][0m
12:35:58     template<typename T, typename T(*func)(Var, ScriptContext*)> bool MemsetConversion(Var value, ScriptContext* scriptContext, T* result)
```

**Comment by MSLaguana** [comment]
@dotnet-bot test OSX static_osx_osx_debug please

**Comment by leirocks** [comment]
fix build break in clang:
template<typename T, typename T(*func)(Var, ScriptContext*)> 
 ==> template<typename T, T(*func)(Var, ScriptContext*)>

**Comment by leirocks** [comment]
@dotnet-bot test OSX static_osx_osx_debug please

**Comment by leirocks** [comment]
@dotnet-bot test Ubuntu shared_ubuntu_linux_debug please

**Comment by MSLaguana** [comment]
Looks like the test timeout failure isn't just OSX, it's also happening on linux debug builds:
```
18:33:26 
ERROR: Test timed out!
18:33:26 /mnt/j/w/Microsoft_ChakraCore/release_1.7/static_ubuntu_linux_debug_prtest/out/Debug/ch -WERExceptionSupport -ExtendedErrorStackForTestHost -BaselineMode -forceNative -off:simpleJit -bgJitDelay:0 -dynamicprofileinput:profile.dpl.793 -off:deferparse /mnt/j/w/Microsoft_ChakraCore/release_1.7/static_ubuntu_linux_debug_prtest/test/Function/FuncBody.bug232281.js
```

**Pull request merged**